mediorum/server/serve_health.go

```diff
@@ -1,9 +1,108 @@
+	SPID                      int    `json:"spID"`
+	SPOwnerWallet             string `json:"spOwnerWallet"`
+	Git                       string `json:"git"`
+	AudiusDockerCompose       string `json:"audiusDockerCompose"`
+	StoragePathUsed           uint64 `json:"storagePathUsed"` // bytes
+	StoragePathSize           uint64 `json:"storagePathSize"` // bytes
+	DatabaseSize              uint64 `json:"databaseSize"`    // bytes
+	AutoUpgradeEnabled        bool   `json:"autoUpgradeEnabled"`
+	SelectedDiscoveryProvider string `json:"selectedDiscoveryProvider"`
+
+	StartedAt         time.Time                  `json:"startedAt"`
+	TrustedNotifier   *ethcontracts.NotifierInfo `json:"trustedNotifier"`
+	Env               string                     `json:"env"`
+	Self              Peer                       `json:"self"`
+	Peers             []Peer                     `json:"peers"`
+	Signers           []Peer                     `json:"signers"`
+	ReplicationFactor int                        `json:"replicationFactor"`
+	Dir               string                     `json:"dir"`
+	ListenPort        string                     `json:"listenPort"`
+	UpstreamCN        string                     `json:"upstreamCN"`
+	TrustedNotifierID int                        `json:"trustedNotifierId"`
+}
+
+type legacyHealth struct {
+	Version                   string `json:"version"`
+	Service                   string `json:"service"`
+	SelectedDiscoveryProvider string `json:"selectedDiscoveryProvider"`
+}
+
+func (ss *MediorumServer) serveHealthCheck(c echo.Context) error {
+	legacyHealth, err := ss.fetchCreatorNodeHealth()
+	data := healthCheckResponseData{
+		Healthy:                   err == nil,
+		Version:                   legacyHealth.Version,
+		Service:                   legacyHealth.Service,
+		SelectedDiscoveryProvider: legacyHealth.SelectedDiscoveryProvider,
+		SPID:                      ss.Config.SPID,
+		SPOwnerWallet:             ss.Config.SPOwnerWallet,
+		Git:                       ss.Config.GitSHA,
+		AudiusDockerCompose:       ss.Config.AudiusDockerCompose,
+		StoragePathUsed:           ss.storagePathUsed,
+		StoragePathSize:           ss.storagePathSize,
+		DatabaseSize:              ss.databaseSize,
+		AutoUpgradeEnabled:        ss.Config.AutoUpgradeEnabled,
+		StartedAt:                 ss.StartedAt,
+		TrustedNotifier:           ss.trustedNotifier,
+		Dir:                       ss.Config.Dir,
+		ListenPort:                ss.Config.ListenPort,
+		UpstreamCN:                ss.Config.UpstreamCN,
+		Peers:                     ss.Config.Peers,
+		Signers:                   ss.Config.Signers,
+		ReplicationFactor:         ss.Config.ReplicationFactor,
+		Env:                       ss.Config.Env,
+		Self:                      ss.Config.Self,
+	}
+	sortedData, err := signature.SortKeys(data)
+	if err != nil {
+		return errors.New("failed to sort health check data: " + err.Error())
```

Comment:
Should errors returned here be `c.JSON(4xx`?

Reply:
yeah that's a good call. changing now